### PR TITLE
Add `TextToImage`, `StableDiffusion3Backbone` and `StableDiffusion3TextToImage`

### PR DESCRIPTION
**Pull request by james77777778**
This is more of a draft, as we may need further discussion regarding the implementation.

Notes for reviewing:
- There are several small modifications in this PR to ensure the numerical stability of all modules/layers. (e.g., LN and Softmax should run in float32)
- `StableDiffusion3Backbone` is a large model that includes all necessary submodules, resulting in a very long init signature. Is this acceptable? How could we refactor it?
- I have figured out how to drop T5 (another 5B model): simply skip it and zero-pad the embeddings from CLIP models.
- Some ideas were borrowed from https://github.com/huggingface/diffusers , which helped simplify the implementation.
- Defining a functional model in `StableDiffusion3Backbone` or `StableDiffusion3TextToImage` is challenging for me. It may be unnecessary for pure inference purposes.
- I couldn't compile the entire `text_to_image` function due to unexpected OOM issues. However, when I split it into `encode`, `denoise` and `decode` functions, it worked fine.
- I have written a rough but functional script that can convert the weights directly from https://huggingface.co/stabilityai/stable-diffusion-3-medium. Please refer to the colab.

Demo colab:
https://colab.research.google.com/drive/1rrQMs0nlKSEzYNhIJChQwgnrZNiydexS?usp=sharing

|`"a cat holding a sign that says hello world"`|`"cute wallpaper art of a cat"`|
|-|-|
|||

TODO:
- [ ] Rename model folder to `stable_diffusion_3`
- [ ] Add docstrings
- [ ] Add weight conversion script
- [ ] Add tests

@divyashreepathihalli @mattdangerw @SamanehSaadat 

BTW, I will be unavailable from 9/17~9/22

### CODE REVIEW

**Comment by james77777778** [comment]
The committee history seems chaotic. I will try to rebase it today.

**Comment by james77777778** [comment]
Since the keras-hub branch was force-pushed, I was unable to rebase it. Therefore, I submitted a new PR for SD3:
https://github.com/keras-team/keras-nlp/pull/1820
